logstash-core/lib/logstash/runner.rb

```diff
@@ -158,32 +176,34 @@ def execute
     if LogStash::Util::JavaVersion.bad_java_version?(java_version)
       $stderr.puts "Java version 1.8.0 or later is required. (You are running: #{java_version})"
       return 1
-    end  
+    end
 
-    LogStash::ShutdownWatcher.unsafe_shutdown = unsafe_shutdown?
+    LogStash::ShutdownWatcher.unsafe_shutdown = setting("pipeline.unsafe_shutdown")
     LogStash::ShutdownWatcher.logger = @logger
 
-    configure
+    configure_plugin_paths(setting("plugin.paths"))
 
     if version?
       show_version
       return 0
     end
 
-    return start_shell(@ruby_shell, binding) if @ruby_shell
+    return start_shell(setting("interactive"), binding) if setting("interactive")
 
-    if config_string.nil? && config_path.nil?
+    @settings.format_settings.each {|line| @logger.log(line) }
```

Comment:
If i understand this properly, this means we log all settings every time we run logstash? is this not doing too much?